acceptance/terrafarm/farmer.go

```diff
@@ -203,6 +203,16 @@ func (f *Farmer) Exec(i int, cmd string) error {
 	return nil
 }
 
+// ExecWriter executes the given command on the i-th writer, returning
+// (in that order) stdout, stderr and an error.
+func (f *Farmer) ExecWriter(i int, cmd string) error {
+	stdout, stderr, err := f.ssh(f.Writers()[i], f.defaultKeyFile(), cmd)
```

Comment:
afaics the only difference here is `s/Writers/Nodes/`, so I think code sharing would make sense.